From fb5035bdc04b2f60f353a5e6a418b402ed89e3b2 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Mon, 6 Dec 2021 11:59:13 +0530
Subject: [PATCH 123/150] net: ethernet: icssg-prueth: Fix release of iep0 on
 probing error

A typo caused crashes in case icss_iep_get_idx succeeded for iep0 but
failed for iep1.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 drivers/net/ethernet/ti/icssg_prueth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ti/icssg_prueth.c b/drivers/net/ethernet/ti/icssg_prueth.c
index 74ba062888a4..3b6c8dd35e85 100644
--- a/drivers/net/ethernet/ti/icssg_prueth.c
+++ b/drivers/net/ethernet/ti/icssg_prueth.c
@@ -2877,7 +2877,7 @@ static int prueth_probe(struct platform_device *pdev)
 	prueth->iep1 = icss_iep_get_idx(np, 1);
 	if (IS_ERR(prueth->iep1)) {
 		ret = dev_err_probe(dev, PTR_ERR(prueth->iep1), "iep1 get failed\n");
-		icss_iep_put(prueth->iep1);
+		icss_iep_put(prueth->iep0);
 		prueth->iep0 = NULL;
 		prueth->iep1 = NULL;
 		goto free_pool;
-- 
2.34.1

